clang/lib/CIR/CodeGen/CIRGenClass.cpp

```diff
@@ -1109,7 +1109,17 @@ void CIRGenFunction::buildDestructorBody(FunctionArgList &Args) {
   // in fact emit references to them from other compilations, so emit them
   // as functions containing a trap instruction.
   if (DtorType != Dtor_Base && Dtor->getParent()->isAbstract()) {
-    llvm_unreachable("NYI");
+    builder.create<mlir::cir::TrapOp>(getLoc(Dtor->getLocation()));
```

Comment:
I'm not sure if this is the best location to use, but I'm also not sure if it matters very much.

Reply:
If `Dtor->getBody()` is available, getting the loc of that stmt would be the best effort.